test/functional-tests/pageObjects/basePage.js

```diff
@@ -9,88 +9,32 @@ class BasePage {
             let EC = protractor.ExpectedConditions;
             browser.wait(EC.visibilityOf(element(locator)), );
         }
+
+        this._formFields = 'ccd-case-edit-form > div > * > *';
+        this._optionalClass = 'ng-valid';
+        this._mandetoryClass = 'ng-invalid';
     }
 
     async waitForElementToBeVisible(element){
-        await browser.wait(EC.visibilityOf(element), TIMEOUT);
+        await browser.wait(await EC.visibilityOf(element), TIMEOUT);
```

Comment:
Not major, but can we make it much graceful with a try, catch. Rest looks good, will not hold off this PR anyway